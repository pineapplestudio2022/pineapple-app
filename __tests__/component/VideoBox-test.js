import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {render} from '@testing-library/react-native';
import VideoBox from '../../src/Components/VideoBoxComponent';

const wrapper = ({children}) => (
  <NativeBaseProvider
    initialWindowMetrics={{
      frame: {x: 0, y: 0, width: 0, height: 0},
      insets: {top: 0, left: 0, right: 0, bottom: 0},
    }}>
    {children}
  </NativeBaseProvider>
);

describe('[VideoBox] Test', () => {
  test('VideoBox snapshot', () => {
    jest.useFakeTimers();
    const rendered = render(<VideoBox />, {wrapper}).toJSON();
    expect(rendered).toMatchSnapshot(); //기존 스냅샷과 일치하는지
    expect(rendered).toBeTruthy(); //컴포넌트가 null, undefinded 같이 falsy 한 값을 가지지 않았는지
  });
});
