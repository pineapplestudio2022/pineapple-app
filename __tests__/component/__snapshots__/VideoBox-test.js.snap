// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[VideoBox] Test VideoBox snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {},
        Object {
          "borderRadius": 8,
          "height": 234.20166073546855,
          "overflow": "hidden",
          "width": 615.3846153846154,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "#595d62",
          "height": "100%",
          "opacity": 0.8,
          "width": "100%",
        }
      }
    >
      <Image
        alt={
          Object {
            "testUri": "../../../src/Assets/Image/image_singing_dumpimage.jpg",
          }
        }
        onError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://img.youtube.com/vi/undefined/mqdefault.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "backgroundColor": "#1a1b1c80",
            "bottom": 0,
            "height": 82.2870699881376,
            "justifyContent": "center",
            "position": "absolute",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        color="#4be3ac"
        fontFamily="body"
        fontSize={29.83103459486446}
        fontStyle="normal"
        fontWeight="bold"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#4be3ac",
              "fontSize": 29.83103459486446,
              "fontStyle": "normal",
              "fontWeight": "700",
              "textAlign": "right",
              "width": "95%",
            },
          ]
        }
        textAlign="right"
        w="95%"
      />
      <Text
        color="#fafafa"
        fontFamily="body"
        fontSize={29.83103459486446}
        fontStyle="normal"
        fontWeight={400}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#fafafa",
              "fontSize": 29.83103459486446,
              "fontStyle": "normal",
              "fontWeight": "400",
              "textAlign": "right",
              "width": "95%",
            },
          ]
        }
        textAlign="right"
        w="95%"
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
